propertyestimator/layers/reweighting.py

```diff
@@ -42,24 +90,188 @@ def schedule_calculation(calculation_backend, storage_backend, layer_directory,
+            systems of the same composition as the desired property.
+        existing_force_fields: :obj:`dict` of :obj:`str` and :obj:`str`
+            A path to a dictionary of all of the force field parameters referenced by the
+            `existing_data`, which have been serialized with `serialize_force_field`
+        """
+        property_class = physical_property.__class__
+
+        if not hasattr(property_class, 'reweight'):
+            return None
+
+        existing_data = {}
+        existing_force_fields = {}
+
+        for substance_id in existing_data_paths:
+
+            existing_data[substance_id] = []
+
+            for data_path in existing_data_paths[substance_id]:
+
+                with open(data_path, 'rb') as file:
+                    existing_data[substance_id].append(pickle.load(file))
+
+        for force_field_id in existing_force_field_paths:
+
+            with open(existing_force_field_paths[force_field_id], 'rb') as file:
+                existing_force_fields[force_field_id] = deserialize_force_field(pickle.load(file))
+
+        reweighted_property = property_class.reweight(physical_property, options, parameter_set_id, existing_data,
+                                                      existing_force_fields, available_resources)
+
+        return_object = CalculationLayerResult()
+        return_object.property_id = physical_property.id
+
+        return_object.calculated_property = reweighted_property
+        return_object.data_to_store = []
+
+        return return_object
+
+    @staticmethod
+    def get_reduced_potential(substance, thermodynamic_state, target_force_field_id,
+                              target_force_field, existing_data, available_resources):
+        """Get the reduced potential from an existing data set. If the target force field
+        id does not match the force field id used by the existing data, the reduced potential
+        will be re-evaluated using the target force field.
+
+        Parameters
+        ----------
+        thermodynamic_state: The thermodynamic state to use when reducing the
+            energies.
+        target_force_field_id: str
+            The id of the target force field parameters.
+        target_force_field: ForceField
+            The target set of force field parameters.
+        existing_data: StoredSimulationData
+            The reference data to get the energies from.
+
+        Returns
+        -------
+        :obj:`numpy.ndarray`
+            The reduced potential of the target state.
+        """
+
+        # If the parameters haven't changed, we only need to swap out the T / P?
+        potential_energies = existing_data.statistics_data.get_observable(ObservableType.PotentialEnergy)
+
+        if target_force_field_id != existing_data.parameter_set_id:
+
+            potential_energies = ReweightingLayer.resample_data(substance, existing_data.trajectory_data,
```

Comment:
Most of the code to do this robustly and expressively already exists. Suggests you might look carefully at @andrrizzi's [`openmmtools.states.ThermodynamicStates`](https://openmmtools.readthedocs.io/en/0.17.0/api/generated/openmmtools.states.ThermodynamicState.html#openmmtools.states.ThermodynamicState) which does most everything you need for computing reduced potentials correctly with OpenMM.

I'd also highly suggest you integrate the [`openmmtools.mcmc`](https://openmmtools.readthedocs.io/en/0.17.0/mcmc.html) module to do all your sampling, and use [`openmmtools.states.SamplerState`](https://openmmtools.readthedocs.io/en/0.17.0/api/generated/openmmtools.states.SamplerState.html#openmmtools.states.SamplerState) to manipulate and store any snapshot data. This would allow us to do more clever things with sampling in the future without needing to make major changes to the code, since we can simply compose MCMC blocks for this purpose.

If you find anything missing from [`openmmtools`](https://openmmtools.readthedocs.io/en/0.17.0/), feel free to raise an issue, or even contribute! We should try to move any generally reusable code---maybe even your pipeline code---there if there's a chance it could be of use to multiple projects!